cognite_toolkit/_cdf_tk/prototypes/commands/dump_assets.py

```diff
@@ -188,6 +191,57 @@ def _buffer(self, asset_iterator: Iterator[tuple[str, list[dict[str, Any]]]]) ->
             if not df.empty:
                 yield group, df
 
+    @lru_cache
+    def get_assets_choice_count_by_dataset(self, item_id: int, client: CogniteClient) -> int:
+        """Using LRU decorator w/o limit instead of another lookup map."""
+        return client.assets.aggregate_count(advanced_filter=Equals("dataSetId", item_id))
+
+    @lru_cache
+    def get_asset_choice_count_by_root_id(self, item_id: int, client: CogniteClient) -> int:
+        """Using LRU decorator w/o limit instead of another lookup map."""
+        return client.assets.aggregate_count(advanced_filter=Equals("rootId", item_id))
+
+    @lru_cache
+    def get_choice_count(self, item_id: int, item: Asset | DataSetWrite, client: CogniteClient) -> int:
+        """Using LRU decorator w/o limit instead of another lookup map."""
+        if isinstance(item, DataSetWrite):
+            return client.time_series.aggregate_count(advanced_filter=Equals("dataSetId", item_id))
+        elif isinstance(item, Asset):
+            return client.assets.aggregate_count(advanced_filter=Equals("dataSetId", item_id))
+        else:
+            raise TypeError(f"Unsupported item type: {type(item)}")
+
+    def _create_choice(self, item_id: int, item: Asset | DataSetWrite, client: CogniteClient) -> questionary.Choice:
+        """
+        Choice with `title` including name and external_id if they differ.
+        Adding `value` as external_id for the choice.
+        `item_id` and `item` came in separate as item is DataSetWrite w/o `id`.
+        """
+
+        ts_count = None
+        if isinstance(item, DataSetWrite):
+            ts_count = self.get_assets_choice_count_by_dataset(item_id, client)
+        elif isinstance(item, Asset):
+            ts_count = self.get_asset_choice_count_by_root_id(item_id, client)
+        else:
+            raise TypeError(f"Unsupported item type: {type(item)}")
+
+        return questionary.Choice(
+            title=f"{item.name} ({item.external_id}) [{ts_count:,}]"
+            if item.name != item.external_id
+            else f"({item.external_id}) [{ts_count:,}]",
+            value=item.external_id,
+        )
+
+    def _get_choice_title(self, choice: str | questionary.Choice | dict[str, Any]) -> str:
+        """
+        Accommodates for the fact that the choice, string or a dict, when `sorted(key=..)` is called.
+        So assert confirm the type of the choice and choice.title and then return the title.
+        """
+        assert isinstance(choice, questionary.Choice)
+        assert isinstance(choice.title, str)  # minimum external_id is set
+        return choice.title.lower()
```

Comment:
Maybe worth a casefold in case of international characters? 

```suggestion
        return choice.title.casefold()
```

Example:
![image](https://github.com/user-attachments/assets/08920873-9c5a-4c39-9eb4-5b2059f297bf)

Reply:
nice, didn't even know it exists!

Reply:
would be interesting how it works with Japan projects 🤔 

asked copilot:
> Japanese characters do not have uppercase or lowercase forms.
> Therefore, applying [str.casefold()](vscode-file://vscode-app/c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) to a Japanese string will not change the string. It will return the string as is.
```
japanese_string = "こんにちは"
casefolded_string = japanese_string.casefold()

print(casefolded_string)  # Output: こんにちは
```